mobile/native/src/ln_dlc/mod.rs

```diff
@@ -525,6 +532,136 @@ pub fn get_usable_channel_details() -> Result<Vec<ChannelDetails>> {
+                if matches!(dlc_channel.state, SubChannelState::Signed(_)) {
+                    tracing::warn!("Trying to recover position from order");
+                    match order::handler::order_filled() {
+                        Ok(order) => {
+                            let (accept_collateral, expiry_timestamp) = node
+                                .inner
+                                .get_collateral_and_expiry_from(dlc_channel.channel_id)?;
+
+                            position::handler::update_position_after_dlc_creation(
+                                order,
+                                accept_collateral,
+                                expiry_timestamp,
+                            )?;
+
+                            tracing::info!("Successfully recovered position from order.");
+                        }
+                        Err(e) => {
+                            tracing::error!("Could not recover position from order as no filling order was found! Error: {e:#}");
+                        }
+                    }
+                } else if matches!(
+                    dlc_channel.state,
+                    SubChannelState::Offered(_)
+                        | SubChannelState::Accepted(_)
+                        | SubChannelState::Finalized(_)
+                ) {
+                    recover_dlc().await?;
+                } else {
+                    tracing::warn!("The DLC is in a state that can not be recovered.");
+                }
+            }
+        }
+        _ => {}
+    }
+
+    Ok(())
+}
+
+/// Sends a pending RecoverDlc background task notification to the UI, allowing the UI to show a
+/// dialog with a spinner that the DLC protocol is still in progress.
+/// Also triggers the `periodic_check` to process any actions that might have been created after the
+/// channel reestablishment.
+///
+/// fixme(holzeis): We currently use different events for show the recovery of a dlc and the waiting
+/// for an order execution in the happy case (without an restart in between). Those events and
+/// dialogs should be aligned.
+async fn recover_dlc() -> Result<()> {
+    tracing::warn!("It looks like the app was closed while the protocol was not finished yet.");
+    event::publish(&EventInternal::BackgroundNotification(
+        BackgroundTask::RecoverDlc(TaskStatus::Pending),
+    ));
+
+    // fixme(holzeis): We are manually calling periodic check here to speed up the processing of
+    // pending actions. Note, this might also not speed up the process, as the coordinator might
+    // have to resend a message to continue the protocol. This should be fixed in `rust-dlc` and any
+    // pending actions should be processed immediately once the channel is ready instead of
+    // periodically checking if a pending action needs to be sent. Note, pending actions can only
+    // get created on channel reestablishment.
+    tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
why are we waiting 5s here?
the comment is about later part of the code, but this is confusing.

🤔 is this only for the UI?

Reply:
I guess the comment wasn't clear enough 🙈. I am going to adapt it.

The 5 seconds wait is pretty arbitrary, but necessary to wait for the channel reestablishment to finish. However that code shouldn't even be here, but handled in `rust-dlc`. I'll probably also wrap that into a separate commit so we can drop it once its fixed in `rust-dlc.

Note, the channel reestablishment identifies where the protocol stopped and creates the required actions to continue / recover the protocol. Since we are only calling the `periodic_check` every 30 seconds I figured I am speeding that up by immediately calling the `periodic_check` here and thus continue the protocol earlier.